app/models/miq_request_workflow.rb

```diff
@@ -855,20 +855,28 @@ def ems_has_clusters?
     false
   end
 
+  FOLDER_PATH_SEPARATOR  = " / ".freeze
+  DATACENTER_FOLDER_TYPE = "Datacenter".freeze
   def get_ems_folders(folder, dh = {}, full_path = "")
+    path = full_path
     if folder.evm_object_class == :EmsFolder
       if folder.hidden
         return dh if folder.name != 'vm'
       else
-        full_path += full_path.blank? ? folder.name.to_s : " / #{folder.name}"
-        dh[folder.id] = full_path unless folder.type == "Datacenter"
+        path = full_path.dup
+        if path.blank?
+          path << folder.name.to_s
+        else
+          path << FOLDER_PATH_SEPARATOR << folder.name
+        end
+        dh[folder.id] = path unless folder.type == DATACENTER_FOLDER_TYPE
```

Comment:
Same question here.

Reply:
Okay, so turns out the `CONSTANT` isn't necessary, but a `.freeze` if it isn't a constant definitely is.  So with the following:

### What we have currently

```ruby
dh[folder.id] = path unless folder.type == DATACENTER_FOLDER_TYPE

# Total allocated: 2055942638 bytes (22741876 objects)
# 
# allocated objects by gem
# -----------------------------------
#    9665227  pending
#    5076997  manageiq/app
#    3513259  manageiq/lib
#    2512007  activerecord-5.0.7
#     861280  activesupport-5.0.7
#     418737  ancestry-2.2.2
#     278793  activemodel-5.0.7
#     178419  ruby-2.3.3/lib
#     165577  arel-7.1.4
#      52875  manageiq-providers-vmware-e2e2b8029538
#      14424  fast_gettext-1.2.0
#       4115  other
#         75  default_value_for-3.0.5
#         68  concurrent-ruby-1.0.5
#         16  memoist-0.15.0
#          4  rubygems
#          3  manageiq-providers-lenovo-ed197e6bea5b
```

### With a normal string

```ruby
dh[folder.id] = path unless folder.type == "Datacenter"

# Total allocated: 2066619256 bytes (23009605 objects)
# 
# allocated objects by gem
# -----------------------------------
#    9665227  pending
#    5344726  manageiq/app
#    3513259  manageiq/lib
#    2512007  activerecord-5.0.7
#     861280  activesupport-5.0.7
#     418737  ancestry-2.2.2
#     278793  activemodel-5.0.7
#     178419  ruby-2.3.3/lib
#     165577  arel-7.1.4
#      52875  manageiq-providers-vmware-e2e2b8029538
#      14424  fast_gettext-1.2.0
#       4115  other
#         75  default_value_for-3.0.5
#         68  concurrent-ruby-1.0.5
#         16  memoist-0.15.0
#          4  rubygems
#          3  manageiq-providers-lenovo-ed197e6bea5b
```


### With a frozen string (not a constant though)

```ruby
dh[folder.id] = path unless folder.type == "Datacenter".freeze

Total allocated: 2055940416 bytes (22741876 objects)

allocated objects by gem
-----------------------------------
   9665227  pending
   5076997  manageiq/app
   3513259  manageiq/lib
   2512007  activerecord-5.0.7
    861280  activesupport-5.0.7
    418737  ancestry-2.2.2
    278793  activemodel-5.0.7
    178419  ruby-2.3.3/lib
    165577  arel-7.1.4
     52875  manageiq-providers-vmware-e2e2b8029538
     14424  fast_gettext-1.2.0
      4115  other
        75  default_value_for-3.0.5
        68  concurrent-ruby-1.0.5
        16  memoist-0.15.0
         4  rubygems
         3  manageiq-providers-lenovo-ed197e6bea5b
```

### With a non-frozen constant

```ruby
# DATACENTER_FOLDER_TYPE = "Datacenter".freeze
DATACENTER_FOLDER_TYPE = "Datacenter"

# ...

dh[folder.id] = path unless folder.type == DATACENTER_FOLDER_TYPE

Total allocated: 2055902327 bytes (22741876 objects)

allocated objects by gem
-----------------------------------
   9665227  pending
   5076997  manageiq/app
   3513259  manageiq/lib
   2512007  activerecord-5.0.7
    861280  activesupport-5.0.7
    418737  ancestry-2.2.2
    278793  activemodel-5.0.7
    178419  ruby-2.3.3/lib
    165577  arel-7.1.4
     52875  manageiq-providers-vmware-e2e2b8029538
     14424  fast_gettext-1.2.0
      4115  other
        75  default_value_for-3.0.5
        68  concurrent-ruby-1.0.5
        16  memoist-0.15.0
         4  rubygems
         3  manageiq-providers-lenovo-ed197e6bea5b
```

So basically is it is a matter of preference between the 1st, 3rd, and 4th options since they all perform the same.